pkg/streamingpromql/operators/function_over_instant_vector.go

```diff
@@ -24,10 +26,34 @@ type FunctionOverInstantVector struct {
 
 	MetadataFunc   functions.SeriesMetadataFunction
 	SeriesDataFunc functions.InstantVectorFunction
+
+	expressionPosition posrange.PositionRange
 }
 
 var _ types.InstantVectorOperator = &FunctionOverInstantVector{}
 
+func NewFunctionOverInstantVector(
+	inner types.InstantVectorOperator,
+	pool *pooling.LimitingPool,
+	metadataFunc functions.SeriesMetadataFunction,
+	seriesDataFunc functions.InstantVectorFunction,
+	expressionPosition posrange.PositionRange,
```

Comment:
>Fair enough though if you want to omit it until we implement the dependent functions.

Yeah, this is what I'm thinking.

The other idea I had was to pass something (the `Query`? something with just the pool and annotations?) to each operator, so we don't have to pass all this stuff around. My only hesitation with this was I worry about passing the `Query` around because it'd be tempting to access things like the start and end timestamps from there, but that will cause issues for subqueries. Maybe I'm being too paranoid though 🙂

Reply:
I think it's fine passing each thing for now :+1: